plugins/modules/ec2_ami.py

```diff
@@ -214,6 +234,22 @@
         - device_name: /dev/sdb
           no_device: yes
 
+- name: AMI Creation with boot_mode and tpm_support
+  amazon.aws.ec2_ami:
+    name: newtest
+    state: present
+    architecture: x86_64
+    virtualization_type: hvm
+    root_device_name: /dev/sda1
+    device_mapping:
+        - device_name: /dev/sda1
+          snapshot_id: "{{ snapshot_id }}"
+    wait: yes
+    region: us-east-1
+    boot_mode: uefi
+    uefi_data: data_file.bin
```

Comment:
UefiData should be a base64 string, so something like ` "{{ lookup('file', '/etc/fstab')|b64encode }}"`.